### ISSUE TITLE
ShortStack 4.0.0 return empty StructVis files

### ISSUE THREAD

**Issue by sebel76** [MikeAxtell/ShortStack#116]
Hi Micheal,

Thank you for the new release! Curious to see how it will improve the annotation of my miRNA, I am running it on my current dataset.

I detected a bug. I can run ShortSack with no issue or error message. Although that ShortStack return all file, the files in the StructVis are empty and we cannot visualise hairpin structures.

I used ShortStack v4.0.0 install, using Conda, on a MacOS 13.2.1 with a processor 3.6 GHz 10-Core Intel Core i9 and 128 GB 2667 MHz DDR4 memory. 

Is the bug related to my machine or installation?

Thank you to help me!
Best,
Sébastien

**Comment by MikeAxtell** [comment]
Thanks Sébastien,

1. Did you run the test example given in the README? If so, did it produce the expected StrucVis output?
2. Can you share the exact command you issued that resulted in a failure of the strucVis output?
3. Can you share the exact and complete STDOUT/STDERR messages that ShortStack outputted for the run that failed StrucVis?

**Comment by sebel76** [comment]
Hi Michael,

1. Did you run the test example given in the README? If so, did it produce the expected StrucVis output?

Thanks you for your short response! I did not run the test example given in the README... ...
I am going to run it this evening and back to you later on!

2. Can you share the exact command you issued that resulted in a failure of the strucVis output?
- Mapping sRNA: 
ShortStack --genomefile {params} --readfile /Users/sbelanger/data/{wildcards.specie}/srna/clean/fastq/*.fastq.gz --outdir {output} --threads 20 --mmap f --align_only --dicermin 21 --dicermax 24

- Annotate microRNAs: 
ShortStack --genomefile {params} --knownRNAs /Users/sbelanger/data/reference/miRBase22/monocots.fa --bamfile /Users/sbelanger/analysis/evo.phas/srna/{wildcards.specie}/merged_alignments.bam --outdir {output} --threads 20 --mmap f --dicermin 21 --dicermax 24 --dn_mirna

3. Can you share the exact and complete STDOUT/STDERR messages that ShortStack outputted for the run that failed StrucVis?

I did not get any message, it just completed the task ending with:

<<< WARNING >>>
Do not rely on these results alone to annotate new MIRNA loci!
The false positive rate for de novo MIRNA identification is low, but NOT ZERO
Insepct each mirna locus, especially the strucVis output, and see
https://doi.org/10.1105/tpc.17.00851 , https://doi.org/10.1093/nar/gky1141

In case it help. I am running ShortStack in a workflow using Snakemake v7.3.8.

Hopefully it help! I will back later after running the test example on my terminal.

Best,
Sébastien

**Comment by MikeAxtell** [comment]
Thanks. I don't know much about SnakeMake.

Do you have the complete STDOUT/STDERR? You only pasted the last few lines. I'd like to look at the whole thing, if possible.

But let's wait until you do the example run from the README.

I appreciate the bug reports!

**Comment by sebel76** [comment]
Hi Micheal,

I ran the test example. I got the same bug than with my dataset. I look to it more closely and found the solution...
The reason is that I modify the ShortStack code to index with the .csi option. My issue it that the strucVis needs editing accordingly.

Original code:
Line 764:	my $index = $file . '.bai';

Correction:
Line 764:	my $index = $file . '.csi';

Now, it works beautifully!!!

Again, thank you for the new version of your software! I will re-run my experiment and digest results; however, preliminary results looks to greatly improve the miRNA annotation!

Best,
Sébastien
